lib/node_modules/@stdlib/iter/cartesian-square/benchmark/benchmark.js

```diff
@@ -0,0 +1,59 @@
+
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require('@stdlib/bench');
+var pkg = require('./../package.json').name;
+var iterCartesianSquare = require('./../lib');
+
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var iterator;
+	var arr = ['a', 'b', 'c'];
+	var v;
+	var i;
+
+	// Run the benchmark...
+	b.tic();
+	for (i = 0; i < b.iterations; i++) {
+		// Reset the iterator:
+		iterator = iterCartesianSquare(arr);
+		v = iterator.next();
+
+		// Iterate over the Cartesian square:
+		while (!v.done) {
+			if (v.done) {
+				b.fail('should iterate over all elements');
+			}
+			v = iterator.next();
+		}
```

Comment:
This not what we are looking for in a benchmark. We are not interested in measuring how long it takes to exhaust an iterator. We should be more interested in the performance of returning a single iterated value.

You can measure iteration without exhausting the iterator by ensuring that the input array has one more element than the square root of `b.iterations`.